### PR TITLE
Generalise media browse dialog from media player

### LINKED ISSUE #7145: Ability to fire the media browser from any card as an action
## The request
I'd like to be able to display the media browser from a `media_player` entity from any card. Right now, firing the `show-dialog` event with the `dialogTag: 'dialog-media-player-browse'` doesn't work because the media browser is not loaded. If it is loaded first though a standard `media_player` card, it works. I wouldn't want to duplicate the media browser in a custom card.

## The alternatives
None

## Additional information

Something like this below currently doesn't work in a custom-card unless it has been displayed before through a core media_player card. `dialogImport` is omitted as it would require to duplicate all the code from the media browser which is far from ideal.
```ts
  private _handleMediaBrowerEvent(actionConfig: MediaBrowserActionConfig): void {
    fireEvent(this, 'show-dialog', {
      dialogTag: 'dialog-media-player-browse',
      dialogParams: {
        action: 'play',
        entityId: actionConfig.entity_id ? actionConfig.entity_id : this._config?.entity,
        mediaPickedCallback: (pickedMedia: any) => {
          this._hass!.callService('media_player', 'play_media', {
            entity_id: actionConfig.entity_id ? actionConfig.entity_id : this._config?.entity,
            media_content_id: pickedMedia.item.media_content_id,
            media_content_type: pickedMedia.item.media_content_type,
          });
        },
      },
    });
  }
```

### PR DESCRIPTION
**Pull request by rob-mccann**
Allows other components and custom components to prompt for users to select an item from the library.

I'm new to HA and feeling my way through the codebase so please do take a good look and let me know if it makes sense!

## Proposed change

Use case: I'd like to build a custom(?) card that launches a specific xbox game without having to find the content IDs.

The idea is to trigger this component from a new input type in the card configuration UI to fill the config.

Moving it to an event driven form means decouples it from the media player and allows it to be called from anywhere, including custom cards in the future.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [X] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #7145

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by homeassistant** [comment]
Hi @rob-mccann, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;frontend#7467).

Once you do that we will be able to review and accept this pull request.

Thanks!

File: src/components/ha-media-player-browse.ts [hunk 1]
```diff
@@ -22,29 +22,29 @@ import {
 import { classMap } from "lit-html/directives/class-map";
 import { ifDefined } from "lit-html/directives/if-defined";
 import { styleMap } from "lit-html/directives/style-map";
-import { fireEvent } from "../../common/dom/fire_event";
-import { computeRTLDirection } from "../../common/util/compute_rtl";
-import { debounce } from "../../common/util/debounce";
-import type { MediaPlayerItem } from "../../data/media-player";
+import { fireEvent } from "../common/dom/fire_event";
```
Comment (bramkragten) [inline src/components/ha-media-player-browse.ts:25]: Why did you move all these files?

File: src/dialogs/media-browse/show-media-browser-dialog.ts [hunk 2]
```diff
@@ -16,12 +16,5 @@ export const showMediaBrowserDialog = (
...
 ): void => {
-  fireEvent(element, "show-dialog", {
-    dialogTag: "dialog-media-player-browse",
-    dialogImport: () =>
-      import(
-        /* webpackChunkName: "dialog-media-player-browse" */ "./dialog-media-player-browse"
-      ),
-    dialogParams,
-  });
+  fireEvent(element, "hass-media-browse", dialogParams);
```
Comment (bramkragten) [inline src/dialogs/media-browse/show-media-browser-dialog.ts:19]: The media browser dialog should be lazy-loaded.

**Comment by bramkragten** [comment]
Not sure I understand what you are trying to do here, I think all you want is already there. The only thing you can't do right now is import the dialog.

But if it is imported (so after you triggered it from the normal UI), you should be able to use:

```ts
fireEvent(element, "show-dialog", {
    dialogTag: "dialog-media-player-browse",
    dialogParams,
  });
```

File: src/components/ha-media-player-browse.ts (hunk 1 above)
Comment (rob-mccann) [reply src/components/ha-media-player-browse.ts:25]: I moved the dialog-related items to the dialog folder which left no need for the additional `media-player` directory in components so I moved `ha-media-player-browse.ts` up a level.

**Comment by rob-mccann** [comment]
@bramkragten I'm trying to launch the browser from a different, custom card - there's no guarantee that it has previously been triggered.

File: src/dialogs/media-browse/show-media-browser-dialog.ts (hunk 2 above)
Comment (rob-mccann) [reply src/dialogs/media-browse/show-media-browser-dialog.ts:19]: I've updated the event listener to lazy load rather than defer load.

https://github.com/home-assistant/frontend/pull/7467/files#diff-5a8931fe0430bd4d3a2c6064914b55b4120c8de15c5f7a6280c9155cd60804a5R30

File: src/components/ha-media-player-browse.ts (hunk 1 above)
Comment (bramkragten) [reply src/components/ha-media-player-browse.ts:25]: But why? The media browser is related to the media player, so it makes sense to have it all together there?

**Comment by rob-mccann** [comment]
The idea is to decouple the browser from the media player - to allow
browsing for other purposes (like component configuration).

I can certainly move it back if preferred - the location of the file isn’t
as important as the other changes here :)

**Comment by bramkragten** [comment]
I still don't understand what you want to do with the dialog, but this should not be a mixin and should just be handled by the normal dialog logic.

**Comment by rob-mccann** [comment]
I wish to create a new card in lovelace.
The card will display media images.
Clicking the image will launch the media on a media player entity.

Now - this card can be configured via YAML but I'd rather provide a UI that allows the user to select the media to show on the card via the media browser. Hope that makes sense?

File: src/components/ha-media-player-browse.ts (hunk 1 above)
Comment (rob-mccann) [reply src/components/ha-media-player-browse.ts:25]: the idea is to allow the media browser to be used outside of the media player - for example, in a card editor to select media content. I don't feel too strongly about this though as it'll work for this usecase irrespective of location :)

**Comment by rob-mccann** [comment]
@bramkragten I've refactored the PR based on your feedback - do let me know if I've put all this in the right place - notably the mixin.

**Comment by bramkragten** [comment]
This is not the way we handle dialogs, you used the deprecated old way. We also are not going to make our code much more complicated to support custom cards. I'm going to close this.

**Comment by 1ntroduc3** [comment]
This is really something we'd like to have. this way we can call the mediabrowser card in a popup for a specific media player. Please reconsider @bramkragten

**Comment by Quentin-Cloos** [comment]
Hi, I really want to be able to call the media browser popup from a button, I don't use the default media player and I don't use the sidebar, is there any chance that you will reconsider this @bramkragten ? I have searched a lot for this and we are many who really need that feature 🙏 Thanks a lot in advance 😀

**Comment by realroywalker** [comment]
@rob-mccann Did you ever manage to achieve launching the media browser via a custom button/card ? - I'm trying to do the same and have been searching for days but no luck.
I really don't want to have the full media player present on my dashboard just to make use of the media browser button, so keen to hear if you found a solution!

**Comment by rob-mccann** [comment]
> @rob-mccann Did you ever manage to achieve launching the media browser via a custom button/card ?

Unfortunately, I didn't make any more progress after the MR was closed. Apologies!
